src/NovelRT/Utilities/Misc.cpp

```diff
@@ -26,6 +26,25 @@ namespace NovelRT::Utilities {
 
       path[pathLength] = L'\0';
       return path;
+#elif defined(__APPLE__)
+      char *path;
+      uint32_t pathLength = 0;
+      _NSGetExecutablePath(nullptr, &pathLength);
+      path = new char[pathLength];
+      if ((pathLength <= 0) || (pathLength >= PATH_MAX))
+      {
+          return std::filesystem::current_path();
+      }
+      
+      if (_NSGetExecutablePath(path, &pathLength) < 0)
+      {
+          return std::filesystem::current_path();
+      }
+      
+      char *actualPath = realpath(path, nullptr);
+      free(path);
+      
+      return std::string(actualPath).c_str();
 #else
       char path[PATH_MAX + 1];
       auto pathLength = readlink("/proc/self/exe", path, PATH_MAX);
```

Comment:
When using the `readlink()` method on macOS, this returns a directory that is one folder above the actual executable's directory (i.e. "artifacts/build/Debug" instead of "artifacts/build/Debug/samples").

Using Apple's native method in this manner allows us to:
1) find the length of the path, then
2) Get the proper executable path as determined by macOS.

With that being said, there's a possibility to get a symbolic link instead of an actual path, so I used `realpath()` here to ensure an absolute path is always returned as long as `_NSGetExecutablePath()` succeeds.

Reply:
Gotcha.